bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentListImpl.java

```diff
@@ -0,0 +1,185 @@
+            return;
+        }
+
+        QueryBuilder queryBuilder = resourceResolver.adaptTo(QueryBuilder.class);
+        if (queryBuilder == null) {
+            LOG.warn("Query builder was null therefore no query was executed");
+            return;
+        }
+
+        Map<String, String> queryParameterMap = new HashMap<>();
+        queryParameterMap.put("path", parentPath);
+        queryParameterMap.put("type", "dam:Asset");
+        queryParameterMap.put("1_property", JcrConstants.JCR_CONTENT + "/data/cq:model");
+        queryParameterMap.put("1_property.value", modelPath);
+
+        ArrayList<String> allTags = new ArrayList<>();
+        if (tagNames != null && tagNames.length > 0) {
+            allTags.addAll(Arrays.asList(tagNames));
+        }
+
+        if (!allTags.isEmpty()) {
+            // Check for the taggable mixin
+            queryParameterMap.put("2_property", JcrConstants.JCR_CONTENT + "/metadata/" + JcrConstants.JCR_MIXINTYPES);
+            queryParameterMap.put("2_property.value", TagConstants.NT_TAGGABLE);
+            // Check for the actual tags (by default, tag are or'ed)
+            queryParameterMap.put("tagid.property", JcrConstants.JCR_CONTENT + "/metadata/cq:tags");
+            for (int i = 0; i < allTags.size(); i++) {
+                queryParameterMap.put(String.format("tagid.%d_value", i + 1), allTags.get(i));
+            }
+        }
+
+        PredicateGroup predicateGroup = PredicateGroup.create(queryParameterMap);
+        Query query = queryBuilder.createQuery(predicateGroup, session);
+
+        SearchResult searchResult = query.getResult();
+
+        LOG.debug("Query statement: '{}'", searchResult.getQueryStatement());
+
+        // Query builder has a leaking resource resolver, so the following work around is required.
+        ResourceResolver leakingResourceResolver = null;
+        try {
+            // Iterate over the hits if you need special information
+            Iterator<Resource> resourceIterator = searchResult.getResources();
+            while (resourceIterator.hasNext()) {
+                Resource resource = resourceIterator.next();
+                if (leakingResourceResolver == null) {
+                    // Get a reference to QB's leaking resource resolver
+                    leakingResourceResolver = resource.getResourceResolver();
+                }
+
+                DAMContentFragment contentFragmentModel = new DAMContentFragmentImpl(
+                        resource, contentTypeConverter, null, elementNames);
+
+                items.add(contentFragmentModel);
+            }
+        } finally {
+            if (leakingResourceResolver != null) {
+                // Always close the leaking query builder resource resolver
+                leakingResourceResolver.close();
+            }
+        }
+    }
+
+    @NotNull
+    @Override
+    public Collection<DAMContentFragment> getListItems() {
```

Comment:
I would have expected the items to be of type `ContentFragment` (content fragment component model).

Reply:
No, that would be wrong. Both models only share the `DAMContentFragment`.